### DISCUSSION TITLE
Is there a way to block dates?

### DISCUSSION THREAD

**Discussion by MalinEbbesen** [ME-MarvinE/XCalendar#95] [Q&A]
Let's say there are no available appointments at certain dates, is there any way to let the user know this? Gray out certain dates or block them?

**Comment by ME-MarvinE** [comment]
Yes. The only way to customise a day is to set the `DayTemplate` property of the CalendarView. You can set this to a `DataTemplate` containing your own custom view, or you can use a [DayView](https://github.com/ME-MarvinE/XCalendar/wiki/DayView-Properties) to replicate the existing look and functionality. The 'Customising a day' example in the [sample app](https://github.com/ME-MarvinE/XCalendar/wiki/Sample-App) does this and would be useful to look at.

When using a DayView, the essential properties you need to set to replicate existing functionality are the `IsSelected`, `IsToday`, `IsCurrentMonth`, and `IsInvalid` properties. You can view more info of how these states work (at the bottom of the page) as well as all the properties in the DayView at the [DayView Properties](https://github.com/ME-MarvinE/XCalendar/wiki/DayView-Properties) wiki page. The [Event Calendar](https://github.com/ME-MarvinE/XCalendar/wiki/Sample-App) example in the sample app is also something you can reference for appointment-style DayViews.
